dracut/modules.d/99kiwi-lib/kiwi-lib.sh

```diff
@@ -42,22 +42,37 @@ function lookup_disk_device_from_root {
     fi
     for disk_device in $(
         lsblk -p -n -r -s -o NAME,TYPE "${root_device}" |\
-            grep disk | cut -f1 -d ' '
+            grep -E "disk|raid" | cut -f1 -d ' '
     ); do
-        disk_matches=$((disk_matches + 1))
+        # Check if root_device is managed by multipath. If this
+        # is the case prefer the multipath mapped device because
+        # directly accessing the mapped devices is no longer
+        # possible
+        if type multipath &> /dev/null; then
+            for wwn in $(multipath -l -v1 "${disk_device}");do
+                if [ -e "/dev/mapper/${wwn}" ];then
+                    disk_device="/dev/mapper/${wwn}"
+                    echo "${disk_device}"
+                    return
+                fi
+            done
+        fi
+        # Check if root_device is managed by mdadm. If this
+        # is the case prefer the md device over the storage
+        # disks because this would create conflicts otherwise
+        if type mdadm &> /dev/null; then
+            for raid_disk in $(
+                mdadm --detail -Y "${disk_device}" 2>/dev/null |\
+                grep -E "MD_DEVICE_.*_DEV" | cut -f2 -d=
```

Comment:
Sincerly, on the Intel Rapid Storage device, the last entry in $disk_device is /dev/sda, which can't be handled by mdadm. Should be /dev/md*

Reply:
mdadm --detail is called on any disk found. the ones that does not provide md information are not taken into account. The loop only runs if there is md information available and takes this md device and returns. I think if you read not by diff but by the complete code you will realized how things work

Reply:
after all you should always have in mind this code must be generic enough for any disk situation including multipath, so we can't just mess it up or make it only for for raid-disks ;)

Reply:
Oh, excuse my rushing - I didn't see the do-loop changed.
Applying the code block gives at:
```
 mdadm --detail -Y "${disk_device}" 2>/dev/null |\
                grep -E "MD_DEVICE_.*_DEV" | cut -f2 -d=
```
`/dev/sda` and `/dev/sdb` as results.

Maybe it is easier to check, if 
```
mdadm --detail -Y "${disk_device}" 2>/dev/null
```
returns 0 (disk_device=/dev/sd* returns 1). In this case it is a valid md-disk.

Reply:
yes you are right that's much easier. I've updated the code. Thanks